overlay/overrides.nix

```diff
@@ -84,7 +84,12 @@ in rec {
   curl-sys = makeOverride {
     name = "curl-sys";
     overrideAttrs = drv: {
-      propagatedBuildInputs = drv.propagatedBuildInputs or [ ] ++ [ (patchCurl pkgs) ];
+      propagatedBuildInputs = drv.propagatedBuildInputs or [ ] ++ [
+        (patchCurl pkgs)
+        pkgs.pkg-config
+      ];
+    } // lib.optionalAttrs (pkgs.stdenv.hostPlatform != pkgs.stdenv.buildPlatform) {
+      PKG_CONFIG_ALLOW_CROSS = "1";
```

Comment:
Are these lines really necessary? They're just duplicating what we already have for `pkg-config`.

Reply:
Given that `curl-sys` doesn't rely on the `pkg-config` Rust crate but instead on the system `pkg-config` directly, we need to duplicate these options here as well. Is this not the case?

Reply:
`curl-sys` [does list](https://docs.rs/crate/curl-sys/0.4.31+curl-7.70.0/source/Cargo.toml) `pkg-config` as a build dependency so the dependencies specified in the override for `pkg-config` should be propagated correctly.

Let's try removing these lines and see if the build still works.

Reply:
Sure, I'll try that right now. 👍

Reply:
Okay, it looks reasonable to me! I've dropped the old commits and force-pushed. Currently awaiting the build to check if it all works, but either way, thanks for the correction.

Reply:
I'll re-request the review as soon as it passes in CI! 🤞

Reply:
Works! 🎉 Thanks a lot. Should be ready for re-review now.